examples/sdf2d.py

```diff
@@ -1,7 +1,7 @@
 import taichi as ti
 from math import tau
 from renderer_utils import reflect, refract
-ti.init()
+ti.init(ti.opengl)
```

Comment:
```suggestion
ti.init(arch=ti.cpu)
```
Umm, I cannot run this example `sdf2d.py` on my MacOS now if it is specified for `ti.opengl`.

Reply:
Does fixing `ti.info` above solve the issue?

Reply:
It should automatically fall back to cpu on that case.

Reply:
Agreed. 
Btw, do we need to keep `ti.init(arch=xxx)` as uniform as possible in `examples/`?
```
./mgpcg_advanced.py:6:ti.init(default_fp=real, arch=ti.x64, kernel_profiler=True)
./laplace.py:3:ti.init(arch=ti.cpu)
./quadtree.py:4:ti.init(arch=ti.cuda)
./mpm_lagrangian_forces.py:4:ti.init(arch=ti.gpu)
./export_ply.py:6:ti.init(arch=ti.cpu)
./fractal.py:3:ti.init(arch=ti.gpu)
./nbody_oscillator.py:3:ti.init(arch=ti.gpu)
./mpm128.py:4:ti.init(arch=ti.gpu) # Try to run on GPU
./sdf2d.py:4:ti.init()
./mgpcg.py:5:ti.init(default_fp=real, arch=ti.x64, kernel_profiler=True)
./pbf2d.py:8:ti.init(arch=ti.gpu)
./euler.py:9:ti.init(arch=ti.gpu, default_fp=real)
./waterwave.py:4:ti.init(arch=ti.gpu)
./game_of_life.py:4:ti.init(arch=ti.gpu)
./taichi_sparse.py:3:ti.init(arch=ti.cuda)
./minimal.py:3:ti.init()
./taichi_bitmasked.py:4:ti.init(arch=ti.gpu)
./cornell_box.py:7:ti.init(arch=ti.gpu)
./sdf_renderer.py:6:ti.init(arch=ti.gpu)
./stable_fluid.py:21:ti.init(arch=ti.gpu)
./simple_uv.py:4:ti.init()
./taichi_dynamic.py:7:ti.init()
./mpm99.py:3:ti.init(arch=ti.gpu) # Try to run on GPU
./export_videos.py:3:ti.init()
./mpm88.py:3:ti.init(arch=ti.gpu)
./particle_renderer.py:8:ti.init(arch=ti.cuda, device_memory_GB=4)
./mgpcg_advanced.py:6:ti.init(default_fp=real, arch=ti.x64, kernel_profiler=True)
./laplace.py:3:ti.init(arch=ti.cpu)
./quadtree.py:4:ti.init(arch=ti.cuda)
./mpm_lagrangian_forces.py:4:ti.init(arch=ti.gpu)
./export_ply.py:6:ti.init(arch=ti.cpu)
./fractal.py:3:ti.init(arch=ti.gpu)
./nbody_oscillator.py:3:ti.init(arch=ti.gpu)
./mpm128.py:4:ti.init(arch=ti.gpu) # Try to run on GPU
./sdf2d.py:4:ti.init()
./mgpcg.py:5:ti.init(default_fp=real, arch=ti.x64, kernel_profiler=True)
./pbf2d.py:8:ti.init(arch=ti.gpu)
./euler.py:9:ti.init(arch=ti.gpu, default_fp=real)
./waterwave.py:4:ti.init(arch=ti.gpu)
./game_of_life.py:4:ti.init(arch=ti.gpu)
./taichi_sparse.py:3:ti.init(arch=ti.cuda)
./minimal.py:3:ti.init()
./taichi_bitmasked.py:4:ti.init(arch=ti.gpu)
./cornell_box.py:7:ti.init(arch=ti.gpu)
./sdf_renderer.py:6:ti.init(arch=ti.gpu)
./stable_fluid.py:21:ti.init(arch=ti.gpu)
./simple_uv.py:4:ti.init()
./taichi_dynamic.py:7:ti.init()
./mpm99.py:3:ti.init(arch=ti.gpu) # Try to run on GPU
./export_videos.py:3:ti.init()
./mpm88.py:3:ti.init(arch=ti.gpu)
./particle_renderer.py:8:ti.init(arch=ti.cuda, device_memory_GB=4)
```